lib_potion_tracking/data/gm4_potion_tracking-1.0/functions/tick.mcfunction

```diff
@@ -4,7 +4,8 @@
 # run from gm4_potion_tracking-1.0:resolve_tick
 
 # function tag so modules can tag their custom potions (should add the "gm4_potion" tag)
-execute as @e[type=potion,tag=!gm4_potion] run function #gm4_potion_tracking:tag_potion
+execute as @e[type=potion,tag=!gm4_potion_tracking_checked] run function #gm4_potion_tracking:tag_potion
+tag @e[type=potion] add gm4_potion_tracking_checked
```

Comment:
Instead of this second command, the library could add its own function in `#tag_potion` to add the tag (it doesn't matter in which order this happens).